### DISCUSSION TITLE
B2A24S20P

### DISCUSSION THREAD

**Discussion by jaqkar** [syssi/esphome-jk-bms#439] [Q&A] [answered]
Hi, bought a JK_B2A24S20P with hardware version V11.XW and software version V11.26. I am getting conflicting info from JK whether or not on can write over TTL/GPS port. I want to switch the discharge FET on/off under certain conditions. Would appreciate any input here.

**Comment by syssi** [comment]
IMO you should be able using the `jk_bms` component to control the charging/discharging switch using UART-TTL.

**Reply by jaqkar** [reply]
Thanks for the quick reply! Was just wondering what the general consensus was on this before I even try it. JK told me I cannot write to the BMS using TTL :(

**Reply by 23900** [reply]
The main purpose of the GPS port is to prevent theft, so discharging/charging can be turned on and off (remotely via mobile network).

**Reply by syssi** [reply, marked as answer]
This is all I know: https://github.com/syssi/esphome-jk-bms/issues/341

For me it's working. Also for a pretty old BMS version (JK-BD6A17S6P, hw 7.2, sw 7.1.0H).

These are the different rumors I'm aware of:

1. The GPS port is ready-only + write support requires a paid customization of the BMS (**false**)
2. The GPS port cannot be used to write commands because the authorization frame/mechanism is/was unknown (**TRUE!**)
3. Write support was added in recent firmware releases (**false?**) but it's limited to the Mosfet controls (**untested**)

**Reply by jaqkar** [reply]
Ok thank you so much for the info. Guess it is time to do some testing and see if I get lucky.

**Reply by avg-I** [reply]
@jaqkar, so did you get lucky?
Please share. Thanks!
